ui/js/controllers/logviewer.js

```diff
@@ -93,21 +103,27 @@ logViewerApp.controller('LogviewerCtrl', [
                     if (bounds.top) {
                         for (var i = data.length - 1; i >= 0; i--) {
                             // make sure we are inserting at the right place
-                            if ($scope.displayedLogLines[0].index != data[i].index + 1) continue;
+                            if ($scope.displayedLogLines[0].index !== data[i].index + 1) {
+                                continue;
+                            }
                             $scope.displayedLogLines.unshift(data[i]);
                         }
 
                         $timeout(function () {
-                            if (above) removeChunkBelow();
+                            if (above) {
+                                removeChunkBelow();
+                            }
                         }, 100);
                     } else if (bounds.bottom) {
                         var sh = element.scrollHeight;
                         var lines = $scope.displayedLogLines;
 
-                        for (var i = 0; i < data.length; i++) {
+                        for (var j = 0; j < data.length; j++) {
                             // make sure we are inserting at the right place
-                            if (lines[ lines.length - 1 ].index != data[i].index - 1) continue;
-                            $scope.displayedLogLines.push(data[i]);
+                            if (lines[ lines.length - 1 ].index !== data[j].index - 1) {
```

Comment:
Wondering the same about the removal of the legacy spaces on each end of `[ lines.length - 1  ]`